calculator.c

```diff
@@ -0,0 +1,431 @@
+		int32_t* firstNbrInt, int32_t* secondNbrInt) {
+	//pointer to fucntion declaration
+	int32_t (*ptrInputIsValid)(char*, char*, int32_t*, int32_t*) = &inputIsValid;
+	double (*ptrExecuteOperation)(int32_t*, char, int32_t*) = &executeOperation;
+
+
+	if ((*ptrInputIsValid)(firstStr, secondStr, firstNbrInt, secondNbrInt)) {
+		return (*ptrExecuteOperation)(*firstNbrInt, operator, *secondNbrInt);
+	}
+	else {
+		ERROR_CODE = invalidInput;
+		return -1;//doesnt matter
+
+	}
+	
+}
+/**
+ *  This fuction is supposed to print the details of an error in case there is one according to ERROR_CODE, a global int32_t variable.
+ *  @param no parameters, it looks at ERROR_CODE, a global int32_t variable.
+ *  @return void, just prints informations to the user
+ */
+void printError() {
+	switch (ERROR_CODE)
+	{
+	case divisionByZero:
+		printf("Failed: can't divide by 0 \n");
+		break;
+	case intOverflow:
+		printf("Failed: int32_t Overflow\n");
+		break;
+	case invalidInput:
+		printf("Failed: invalid input \nOnly positive int32_t are accepted\n");
+		break;
+	case invalidOperator:
+		printf("Failed: invalid operator \nOnly {+, -, *, /, %, r, ^, :} are supported\n");
+		break;
+	default:
+		printf("Failed\n");
+	}
+}
+
+/**
+ *  This fuction is supposed to print either the result or the details of an error in case there is one.
+ *  In case there is no error the function just allows us to get rid of the .0000 in case of result being int32_t:
+ * for example, in case of 1+2 the result is a doble and will print "3.000", this function prints 3 "instead"
+ *  @param no parameters, it looks at ERROR_CODE, a global int32_t variable.
+ *  @return void, just prints informations to the user
+ */
+void printResult(double result) {
+	//pointer to fucntion declaration
+	void (*ptrPrintError)() = &printError;
+	if (ERROR_CODE) {
+		(*ptrPrintError)();
+	}
+	else if ((int32_t)result == result) {//we geck if restult is a whole number
+		int32_t resultInt = (int32_t)result;
+		printf("%d\n", resultInt);//to get rid of the .0000 in case of result being int32_t
+	}
+	else {
+		printf("%f\n", result);
+	}
+}
+/**
+ *  This is the main() function we call in case we dont want to run tests and want to execute 
+ *  the prgram with our own inputs. It asks for the two nubers and the operator, 
+ 
+ */
+int32_t mainForDemo() {
+	//pointer to fucntion declaration
+	double (*ptrCalculate) (char*, char, char*, int32_t*, int32_t*) = &calculate;
+	void (*ptrPrintResult) (double) = &printResult;
+
+	char* operator = malloc((MAXSIZEINPUT + 1) * sizeof(char));
+	char* firstStr = malloc((MAXSIZEINPUT + 1) * sizeof(char));
+	char* secondStr = malloc((MAXSIZEINPUT + 1) * sizeof(char));
+	int32_t firstNbrInt, secondNbrInt; // these are our int32_t buffers that will host the parsed input 
+	double result = 0;
+
+	printf("Insert first number :");
+	scanf_s("%s", firstStr, MAXSIZEINPUT+1);
+
+	printf("Insert operation: ");
+	scanf_s("%s", operator, MAXSIZEINPUT);

```

Comment:
because scanf_s("%c", &operator, 1); just skipped my input, 
but, I just solved it by writing 
" %c" instead of  "%c"  (space) thanks to https://stackoverflow.com/a/5109604